<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>751</width>
    <height>440</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>40</y>
      <width>151</width>
      <height>191</height>
     </rect>
    </property>
    <property name="title">
     <string>1.Image Processing</string>
    </property>
    <widget class="QPushButton" name="Btn_1_1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1.1 Load Image</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Btn_1_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1.2 Color Conversion</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Btn_1_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1.3 Image Conversion</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Btn_1_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1.4 Blending</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>40</y>
      <width>141</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>2. Adaptive Threshold</string>
    </property>
    <widget class="QPushButton" name="Btn_2_1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>2.1 Global Threshold</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Btn_2_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>2.2 Local Threshold</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>40</y>
      <width>201</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>3. Image Transformation</string>
    </property>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>181</width>
       <height>211</height>
      </rect>
     </property>
     <property name="title">
      <string>3.1 Rot, scale, Translate</string>
     </property>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>161</width>
        <height>141</height>
       </rect>
      </property>
      <property name="title">
       <string>Parameters</string>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>47</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>Angle:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>47</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>Scale:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>47</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>Tx:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>47</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>Ty:</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="Angle_box">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>20</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="Scale_box">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>50</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="Tx_box">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>80</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="Ty_box">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>110</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>20</y>
         <width>47</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>deg</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>80</y>
         <width>47</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>pixel</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>110</y>
         <width>47</width>
         <height>12</height>
        </rect>
       </property>
       <property name="text">
        <string>pixel</string>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="Btn_3_1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>161</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>3.1 Rotation, scaling, translation</string>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="Btn_3_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>240</y>
       <width>161</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>3.2 Perspective Transform</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>150</y>
      <width>151</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>4.Convolution</string>
    </property>
    <widget class="QPushButton" name="Btn_4_1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>1.Gaussian</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Btn_4_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>2. Sobel X</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Btn_4_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>3. Sobel Y</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Btn_4_4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>4. Magnitude</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="Btn_5_1">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>40</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>5.1 Show Train Images</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Btn_5_2">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>80</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>5.2 Show Hyperparameters</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Btn_5_3">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>120</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>5.3 Train 1 Epoch</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Btn_5_4">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>160</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>5.4 Show Training Result</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>210</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Test Image Index</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="Image_index_box">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>210</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="Btn_5_5">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>240</y>
      <width>151</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>5.5 Inference</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>751</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
